src/kio/serial/errors.py

```diff
@@ -10,13 +10,17 @@ class UnexpectedNull(DecodeError):
     ...
 
 
-class SchemaError(SerialError):
+class BufferUnderflow(DecodeError):
```

Comment:
nit: group by type inherited and set to alphabetical order?

Reply:
Decided to do less here for now, so just adding the new exception. Follow-up which applied the proposed ordering here: https://github.com/Aiven-Open/kio/pull/123